### DISCUSSION TITLE
Taking the curl of two variables in xarray

### DISCUSSION THREAD

**Discussion by pecos27** [pydata/xarray#6614] [Q&A] [answered]
General question, but is there a way to take the curl of two 3D arrays (time, lat, and lon) in xarray?
For example if you want to take the wind stress curl from the zonal and meridional wind stress, is there something similar to np.gradient in xarray? 
Or any general recommendations in taking the curl with multidimensional data?
Thanks!

**Comment by andersy005** [comment, marked as answer]
> Or any general recommendations in taking the curl with multidimensional data?

xgcm may provide the core functionality for this 

- See this issue: https://github.com/xgcm/xgcm/issues/187
- This example might be useful too: https://pop-tools.readthedocs.io/en/latest/examples/pop_div_curl_xr_xgcm_metrics_compare.html#curl

**Reply by pecos27** [reply]
Thanks! This is really helpful!

**Reply by TomNicholas** [reply]
There is another example of calculating the curl with xGCM [here too](https://xgcm.readthedocs.io/en/latest/ufunc_examples.html#vorticity).
